csrc/ir/base_nodes.cpp

```diff
@@ -144,7 +146,7 @@ void Val::resolveIndexDtype() {
       index_dtype == DataType::Int || index_dtype == DataType::Int32,
       "Invalid index data type: ",
       index_dtype);
-  dtype_ = index_dtype;
+  dtype_ = DataType::Index;
```

Comment:
Why is it changed to the index type? It seems this function just changes the data type of any integer Val to the index type.

Reply:
It is just a matter of splitting the work. This should be a no-op change. I want to leave the no-op change here, and update all the string compare tests in this PR, and leave the `resolveIndexType()` removal PR as small as possible.